docs/pages/access-controls/access-graph/self-hosted-helm.mdx

```diff
@@ -0,0 +1,220 @@
+
+(!docs/pages/kubernetes-access/helm/includes/helm-repo-add.mdx!)
+
+## Step 2/4. Set up the Teleport Access Graph service
+
+You will need a copy of your Teleport cluster's host certificate authority (CA).
+TAG requires incoming connections to be authenticated via host certificates that the host CA issues to the Auth Service and Proxy Service.
+
+The host CA can be retrieved in one of the following ways:
+
+<Tabs>
+<TabItem label="Via curl">
+```code
+$ curl 'https://<Var name="teleport.example.com" />/webapi/auth/export?type=tls-host'
+-----BEGIN CERTIFICATE-----
+MIIDqjCCApKgAwIBAgIQMIK8/WiQ/rUOrjlmB0IHVTANBgkqhkiG9w0BAQsFADBv
+<...>
+-----END CERTIFICATE-----
+```
+</TabItem>
+
+<TabItem label="Via tctl">
+```code
+$ tsh login --proxy=<Var name="teleport.example.com" />
+$ tctl get cert_authorities --format=json \
+    | jq -r '.[] | select(.spec.type == "host") | .spec.active_keys.tls[].cert' \
+    | base64 -d
+-----BEGIN CERTIFICATE-----
+MIIDqjCCApKgAwIBAgIQMIK8/WiQ/rUOrjlmB0IHVTANBgkqhkiG9w0BAQsFADBv
+<...>
+-----END CERTIFICATE-----
+```
+</TabItem>
+</Tabs>
+
+Then, create the namespace and required secrets for the TAG deployment.
+You can skip the final command if you are instead using cert-manager to issue the certificate,
+or otherwise have already provisioned the TLS certificate for Teleport Access Graph in your Kubernetes cluster.
+
+```code
+$ kubectl create namespace teleport-access-graph 
+$ kubectl -n teleport-access-graph create secret generic teleport-access-graph-postgres \
+    --from-literal uri="postgres://access_graph_user:my_password@db.example.com:5432/access_graph_db?sslmode=require"
+$ kubectl -n teleport-access-graph create secret tls teleport-access-graph-tls --cert=./certs/cert.crt --key=./certs/cert.key
+```
+
+The above commands assume that you have the TLS certificate and private key for the TAG server stored under the `./certs` directory.
+
+<Details title="Generating TLS certificates for development">
+The following script can be used to quickly generate a certificate authority and server TLS certificate for deploying TAG.
+
+```sh
+#!/usr/bin/env bash
+
+set -e
+
+mkdir -p ./certs
+cd ./certs
+
+openssl genrsa -aes256 -out ca.key 4096
+openssl req -x509 -new -key ca.key -sha256 -days 3652 -out ca.crt -subj '/CN=root'
+openssl req -new -out cert.csr -newkey rsa:4096 -nodes -keyout cert.key -subj '/CN=Access Graph'
+openssl x509 -req -in cert.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out cert.crt -days 3652 -sha256 \
+  -extfile <(printf 'extendedKeyUsage = serverAuth, clientAuth\nsubjectAltName = DNS:teleport-access-graph.teleport-access-graph.svc.cluster.local')
+
+echo "Certificates issued successfully"
+```
+
+The script will ask to input the same password 4 times:
+twice to create and confirm the password to encrypt the CA private key material with,
+then once to use the private key to self-sign the CA certificate,
+and finally once to use for signing the TAG server certificate.
```

Comment:
I'm surprised by the password prompt. I have a very similar openssl script, and it doesn't encrypt the CA. Is that because of the `-aes256` flag? Can we just remove it? I don't think we care about encrypting the CA as this is a snake oil script for local dev anyway.

Reply:
I intentionally included the `aes256` flag for the CA cert. This is a snakeoil script, but I still tried to be cautious. We could remove it, it is up to you.

For context, our [HSM guide](https://goteleport.com/docs/choose-an-edition/teleport-enterprise/hsm/#step-15-set-up-your-hsm) includes `aes256`, but it is not positioned as a "snakeoil" script